pkg/collector/py/utils.go

```diff
@@ -181,3 +201,141 @@ func getModuleName(modulePath string) string {
 	// no need to check toks length, worst case it contains only an empty string
 	return toks[len(toks)-1]
 }
+
+func decAllRefs(refs []*python.PyObject) {
+	for _, ref := range refs {
+		ref.DecRef()
+	}
+}
+
+// GetPythonInterpreterMemoryUsage collects python interpreter memory usage
+func GetPythonInterpreterMemoryUsage() ([]*PythonStats, error) {
+	glock := newStickyLock()
+	defer glock.unlock()
+
+	activeRefs := []*python.PyObject{}
+	defer decAllRefs(activeRefs)
+
+	memStatModule := python.PyImport_ImportModule(pyMemModule)
+	if memStatModule == nil {
+		return nil, fmt.Errorf("Unable to import Python module: %s", pyMemModule)
+	}
+	activeRefs = append(activeRefs, memStatModule)
+
+	statter := memStatModule.GetAttrString(pyMemSummaryFunc)
+	if statter == nil {
+		pyErr, err := glock.getPythonError()
+
+		if err != nil {
+			return nil, fmt.Errorf("An error occurred while grabbing the python memory statter: %v", err)
+		}
+		return nil, errors.New(pyErr)
+	}
+	activeRefs = append(activeRefs, statter)
+
+	args := python.PyTuple_New(0)
+	kwargs := python.PyDict_New()
+	activeRefs = append(activeRefs, args)
+	activeRefs = append(activeRefs, kwargs)
+
+	stats := statter.Call(args, kwargs)
+	if stats == nil {
+		pyErr, err := glock.getPythonError()
+
+		if err != nil {
+			return nil, fmt.Errorf("An error occurred collecting python memory stats: %v", err)
+		}
+		return nil, errors.New(pyErr)
+	}
+	activeRefs = append(activeRefs, stats)
+
+	keys := python.PyDict_Keys(stats)
+	if keys == nil {
+		pyErr, err := glock.getPythonError()
+
+		if err != nil {
+			return nil, fmt.Errorf("An error occurred collecting python memory stats: %v", err)
+		}
+		return nil, errors.New(pyErr)
+	}
+	activeRefs = append(activeRefs, keys)
+
+	myPythonStats := []*PythonStats{}
+	var entry *python.PyObject
+	for i := 0; i < python.PyList_GET_SIZE(keys); i++ {
+		entryName := python.PyString_AsString(python.PyList_GET_ITEM(keys, i))
+		entry = python.PyDict_GetItemString(stats, entryName)
+		if entry == nil {
+			// key surprisingly unavailable...
+			continue
+		}
+		activeRefs = append(activeRefs, entry)
```

Comment:
`entry` (returned by `PyDict_GetItemString`) is borrowed, we shouldn't count it in the `activeRefs` (see https://docs.python.org/2/c-api/dict.html#c.PyDict_GetItemString)